### PR TITLE
Add `reload_config()` dispatcher

### PR DESCRIPTION
**Pull request by LRitzdorf**
#### Describe your PR, what does it fix/add?
An `hl.dsp.reload_config()` dispatcher, because shelling out to `hyprctl reload` only to then do IPC back to Hyprland feels awkward.

Wiki PR: hyprwm/hyprland-wiki#1646

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
Nothing super complicated here. The only thing is, we use `doLater` to not tear down config state while it's in active use, but that's pretty routine.

#### Is it ready for merging, or does it need work?
Tested locally, and it reloads properly (wiping out extra Lua variables, for example) without crashing or anything.

### CODE REVIEW

File: src/config/shared/actions/ConfigActions.cpp [hunk 1]
```diff
@@ -1218,6 +1220,12 @@ ActionResult Actions::exit() {
     return {};
 }
 
+ActionResult Actions::reloadConfig() {
+    // probably don't tear down state while stuff is actively running
```
Comment (vaxerski) [inline src/config/shared/actions/ConfigActions.cpp:1226]: very good.

**Review by vaxerski — COMMENTED** [review]
- Needs a test
- Needs to throw an error if reload_config() is called from the global scope (otherwise it will reload forever) - exceptions apply with repl / eval

**Comment by LRitzdorf** [comment]
For the infinite reload thing, I don't thiiiink we can really do Lua scope detection, but a better way IMO is to just make the dispatch fail if the config isn't done being loaded yet. That was easy enough to implement, didn't need carveouts for eval/repl, and even saves you from reloading in the `config.reloaded` handler if you somehow decide that's a good life choice.

What were you thinking in the way of tests? A basic "set var, reload, make sure var got wiped," or something fancier to hit edge cases like the above? If the latter, we'd kinda need a new config file for just that test, which sounds like pain to set up with hyprtester...

**Comment by LRitzdorf** [comment]
Dang layer-pointer-focus test...

**Comment by vaxerski** [comment]
> For the infinite reload thing, I don't thiiiink we can really do Lua scope detection, but a better way IMO is to just make the dispatch fail if the config isn't done being loaded yet. That was easy enough to implement, didn't need carveouts for eval/repl, and even saves you from reloading in the config.reloaded handler if you somehow decide that's a good life choice.

you dont need to, simply pop an error if it's called while the config is being parsed from the main ::reload()

> What were you thinking in the way of tests? A basic "set var, reload, make sure var got wiped," or something fancier to hit edge cases like the above? If the latter, we'd kinda need a new config file for just that test, which sounds like pain to set up with hyprtester...

the first is enough I think

**Comment by LRitzdorf** [comment]
> simply pop an error if it's called while the config is being parsed from the main ::reload()

That's basically what I've done. The ConfigManager interface now includes a `configLoaded()` method, which for Lua just returns false if it's currently parsing (i.e. we're in `::reload()`). Dispatching `reload_config()` checks that and pops an error if it's false.

Also, the test is in here now, so this should be mergeable whenever.

**Review by vaxerski — APPROVED** [review]
